doc/developer/expression-format.md

```diff
@@ -0,0 +1,34 @@
+# Expression Format
+
+This document describes the representations we use for the logic expressions
+that Edsger manipulates.
+
+Expressions are represented as lists. The first element is the operator and the
+operands follow. Operators will be represented as Clojure keywords. The
+following is a list of operations that Edsger should handle properly.
+
+```
+:equiv -- boolean equivalance   -- any number of booleans
+:equal -- equality              -- any number of booleans
+:and   -- logical AND           -- any number of booleans
+:or    -- logical OR            -- any number of booleans
+:not   -- logical NOT           -- one boolean
+:implies -- logical implication -- two booleans
+```
+
+That list should be enough to get us going, but we'll want the following in the
+future.
+
+```
+:greater       -- greater than (>)      -- any number of integers
+:less          -- less than (<)         -- any number of integers
+:element-of    -- self-explanetory name -- two operands, second is a set
+:strict-subset -- subset, but not equal -- two sets
+:subset        --                       -- two sets
```

Comment:
Which ordering do you have in mind for this? (first contains second vs second contains first) If the latter, what do you think about adding '-of' like 'subset-of'? It doesn't really matter at this point, but it was a little unclear to me.

Reply:
I was thinking that `(:subset A B)` would mean that `A` is a subset of `B`. I'll update the doc.